### PR TITLE
FSSDK-8970: Adds support for odp options in config.

### PR DESCRIPTION
**Pull request by yasirfolio3**
## Summary
- Adds support for providing `disableOdp`, `eventsRequestTimeout`, `eventsFlushInterval`, `segmentsCacheSize`, `segmentsRequestTimeout` and `segmentsCacheTimeout` options through config.yaml

## Issues
- https://jira.sso.episerver.net/browse/FSSDK-8970

## Tests
- Added new unit tests.
- All previous tests should pass.

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -135,6 +135,12 @@ client:
     ## By default Agent assumes only alphanumeric characters as part of the SDK Key string.
     ## https://github.com/google/re2/wiki/Syntax
     sdkKeyRegex: "^\\w+(:\\w+)?$"
+    ## Disable odp
+    disableOdp: false
+    ## Max number of items that can be stored in odp segments cache.
+    segmentsCacheSize: 10000
+    ## Timeout in seconds after which an item stored in odp segments cache will be deleted.
+    segmentsCacheTimeout: 600s
```
Comment (jaeopt) [inline config.yaml:143]: What about grouping all these and future odp options under "odp". I think "odpCache" is also can be a part of this group. We can also clarify "odpCache" will override these values.

**Review by jaeopt — CHANGES_REQUESTED** [review]
I have a suggestion for grouping all odp configurations. Let me know if it makes sense.
Other than that, all looks good to me.

File: config.yaml (hunk 1 above)
Comment (Mat001) [reply config.yaml]: What about:
-  odp_segment_request_timeout
- odp_event_request_timeout
- odp_event_flush_interval

We use these in Python sdk ODP sdk settings, but don't see them exposed in go-sdk. Are we including them in agent config?

**Review by Mat001 — CHANGES_REQUESTED** [review]
Made a comment about config options.

File: config.yaml (hunk 1 above)
Comment (yasirfolio3) [reply config.yaml]: Have only added those params which are directly available through the main factory constructor. Should we be adding these internal properties as well? @jaeopt

Comment (yasirfolio3) [reply config.yaml]: @Mat001 request timeout's and flush interval can also be updated in go-sdk but for that we need to supply instances of both the event manager and segment manager to the odp manager.

Comment (jaeopt) [reply config.yaml]: @Mat001 a good call! I think it's valuable to provide those options (segment_request_timeout, event_request_timeout, and probably event_flush_interval as well) to agent customers. @yasirfolio3

File: config.yaml [hunk 2]
```diff
@@ -164,6 +158,13 @@ client:
...
         #     Auth-Token: "12345"
     odp:
+      ## Disable odp
+      disable: false
+      ## Max number of items that can be stored in odp segments cache.
+      segmentsCacheSize: 10000
+      ## Timeout in seconds after which an item stored in odp segments cache will be deleted.
+      segmentsCacheTimeout: 600s
+      ## This will override all other settings
       cache:
```
Comment (jaeopt) [inline config.yaml:168]: Looks good! What about "segmentsCache" instead of "cache" to be consistent with "segmentsCacheSize" and "segmentsCacheTimeout"?

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks good! A name change suggested. Can we also add timeouts and flush-interval to odp options as @Mat001 suggested?

File: config.yaml (hunk 2 above)
Comment (yasirfolio3) [reply config.yaml:168]: sure, will update this in the parent PR for odp-cache, this is branched off of it.

**Review by jaeopt — APPROVED** [review]
LGTM

**Review by Mat001 — APPROVED** [review]
Looks good with the additional config options.

**Pull request merged**